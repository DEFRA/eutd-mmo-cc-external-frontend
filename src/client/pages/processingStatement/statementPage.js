import React, { Fragment } from 'react';
import { withRouter } from 'react-router';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { Header, Panel, Main, GridRow, GridCol } from 'govuk-react';

import {
  clearProcessingStatement,
  saveProcessingStatement,
  saveProcessingStatementToRedis
} from '../../actions';

import {
  getCompletedDocument
} from '../../actions/document.actions';

import PageTitle from '../../components/PageTitle';
import PageTemplateWrapper from '../../components/PageTemplateWrapper';
import HelpLink from '../../components/HelpLink';
import { monitorEvent } from '../../actions/monitor.actions';
import {withTranslation} from 'react-i18next';

class StatementPage extends React.Component {

  hasRequiredData() {
    const { completedDocument } = this.props;

    return completedDocument && completedDocument.documentNumber && completedDocument.documentUri && completedDocument.documentNumber.includes('-PS-');
  }

  async componentDidMount() {
    const documentNumber = this.props.match.params['documentNumber'];

    await this.props.getCompletedDocument(documentNumber);

    if (!this.hasRequiredData()) {
      this.props.history.push('/create-processing-statement/processing-statements');
    }
  }

  render() {
    const {route, completedDocument, t} = this.props;
    const documentNumber = (completedDocument && completedDocument.documentNumber) ? completedDocument.documentNumber : 'not found';
    const uri = (completedDocument && completedDocument.documentUri) ? completedDocument.documentUri :  'not found';

    const buildUrl = (docUrl) => {
      if(docUrl.indexOf('/') !== -1) {
        const splitUrlLength = docUrl.split('/').length;
        const splitUrl = docUrl.split('/')[splitUrlLength - 1];
        const docId = splitUrl.split('?')[0];
        return `/pdf/export-certificates/${docId}`;
      }
      return `/pdf/export-certificates/${docUrl}`;
    };

    const monitorViewEvent = doc => {
      const journey = this.props.route.journey;
      this.props.monitorEvent(doc, journey);
    };

    return (
      <Main>
        <PageTitle title={t('psCreatedSuccessPageTitle')} />
        <GridRow>
          <GridCol>
            <Panel
              className="submitted-certificate-bg"
              panelTitle={t('psCreatedPanelTitle')}
              panelBody={
                <Fragment>
                  {t('psCreatedPanelBody')}
                  <br />
                  <strong id="documentNumber">{documentNumber}</strong>
                </Fragment>
              }
            />
          </GridCol>
        </GridRow>
        <GridRow>
          <GridCol>
            <Header level={2} size="MEDIUM">{t('psCreatedNextStepsHeader')}</Header>
            <ol className="list list-number">
              <li>
                <strong><a rel="noopener noreferrer" href={buildUrl(uri)} onClick={monitorViewEvent.bind(this, documentNumber)}>{t('psCreatedDownloadLink')}</a>. </strong>
                {t('psCreatedDownloadDocumentNotesSubHeading')}
                <ul className="list list-bullet">
                  <li>{t('psCreatedDownloadDocumentNotesFirefox')}</li>
                  <li>{t('psCreatedDownloadDocumentNotesMobile')}</li>
                </ul>
              </li>
              <li>
                <strong>{t('psCreatedEmailToImporter')}</strong>
                {t('commonDocumentCreatedImportersResponsibility')}
              </li>
            </ol>
            <p>
              <Link to={'/create-processing-statement/processing-statements'}>{t('psCreatedViewOrCreateNewLink')}</Link>
            </p>
            <HelpLink journey={route.journey}/>
          </GridCol>
        </GridRow>
      </Main>
    );
  }
}

function mapStateToProps(state) {
  return {
    processingStatement: state.processingStatement,
    documents: state.documents,
    monitorEvent: state.monitorEvent,
    completedDocument: state.completedDocument
  };
}

function loadData(store) {
  if (Object.keys(this.queryParams).length !== 0) {
    store.dispatch(clearProcessingStatement());
    store.dispatch(saveProcessingStatement({uri: this.queryParams.uri, documentNumber: this.queryParams.documentNumber}));
  }
}

export const component = withRouter(
  connect(
    mapStateToProps,
    {
      save: saveProcessingStatement,
      saveToRedis: saveProcessingStatementToRedis,
      monitorEvent: monitorEvent,
      clear: clearProcessingStatement,
      getCompletedDocument: getCompletedDocument
    }
  )(withTranslation()(StatementPage))
);

export default {
  loadData,
  component: PageTemplateWrapper(component)
};
