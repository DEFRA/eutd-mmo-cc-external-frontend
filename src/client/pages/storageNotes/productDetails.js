import React from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import _ from 'lodash';
import SpeciesAutocomplete from '../../components/SpeciesAutocomplete.js';

import { Main, GridRow, GridCol, Header, BackLink, InputField } from 'govuk-react';

import PageTemplateWrapper from '../../components/PageTemplateWrapper';
import {
  toGovukErrors,
  onHandleErrorClick,
  scrollToErrorIsland
} from '../utils';
import ErrorIsland from '../../components/elements/ErrorIsland';
import PageTitle from '../../components/PageTitle';
import Form from '../../components/elements/Form';
import {
  getStorageNotesFromRedis,
  saveStorageNotes,
  saveStorageNotesToRedis,
  clearStorageNotes
} from '../../actions';

import SpeciesDetails from '../../components/elements/SpeciesDetails';
import HelpLink from '../../components/HelpLink';
import SaveAsDraftButton from '../../components/SaveAsDraftButton';
import ContinueButton from '../../components/elements/ContinueButton';
import { getAllFish } from '../../actions';
import DateFieldWithPicker from '../../components/DateFieldWithPicker.js';
import { withTranslation } from 'react-i18next';
import BackToProgressLink from '../../components/BackToProgressLink';
class CatchDetailsPage extends React.Component {

  state = {};

  constructor(props) {
    super(props);
    this.setSpecies = this.setSpecies.bind(this);
    this.state = this.init(props.storageNotes, props);
  }

  init(data, props) {
    const storageNotes = _.cloneDeep(data || {});
    const index = +(props.match.params.productIndex) || 0;
    if (storageNotes.catches && index >= storageNotes.catches.length) {
      storageNotes.catches.push({}); // add new entry if needed
    }

    return ({
      index,
      storageNotes,
      errors: storageNotes.errors || {},
      species: '',
      scientificName: '',
      selectedDateOfUnloading: storageNotes.catches && storageNotes.catches[index] ? storageNotes.catches[index].dateOfUnloading : ''
    });

  }

  redirectToForbidden() {
    this.props.history.push('/forbidden');
  }
  componentDidUpdate() {
    const { storageNotes } = this.props;

    if (storageNotes.unauthorised === true) {
      this.redirectToForbidden();
    }
  }

  async componentDidMount() { // js land
    window.scrollTo(0, 0);

    const { documentNumber } = this.props.match.params;
    let data = await this.props.getFromRedis(this.props.match.params.documentNumber) || {};

    this.setState(this.init(data, this.props));

    const { index, storageNotes } = this.state;
    const product = storageNotes.catches[index];

    if (product === undefined) {
      this.props.history.push(`/create-storage-document/${documentNumber}/you-have-added-a-product`);
    }

    await Promise.all([
      this.props.getAllFish()
    ]);
  }

  componentWillUnmount() {
    this.props.clear();
  }

  async save(nextUri) {
    const { index } = this.state;
    const { documentNumber } = this.props.match.params;

    const currentUrl = this.props.route.firstProduct
      ? `/create-storage-document/${documentNumber}/add-product-to-this-consignment`
      : `/create-storage-document/${documentNumber}/add-product-to-this-consignment/${index}`;

    const { saveAsDraftUri } = this.props.route;
    const isCatchDetailsSavedAsDraft = nextUri === saveAsDraftUri ? true : false;
    const storageNotes = await this.props.saveToRedis({
      saveToRedisIfErrors: isCatchDetailsSavedAsDraft,
      data: this.state.storageNotes,
      currentUrl: currentUrl,
      documentNumber: documentNumber
    });

    if (!isCatchDetailsSavedAsDraft && !_.isEmpty(storageNotes.errors)) {
      this.setState({
        errors: storageNotes.errors,
        errorsUrl: storageNotes.errorsUrl
      });
      scrollToErrorIsland();
      setTimeout(
        () =>
          document
            .getElementById('errorIsland')
            .scrollIntoView({ behavior: 'smooth' }),
        0
      );
      return;
    }

    this.props.history.push(nextUri);
  }

  onSaveAsDraft = async (e) => {
    e.preventDefault();
    const { saveAsDraftUri } = this.props.route;
    await this.save(saveAsDraftUri);
  }

  goBack(e, page) {
    e.preventDefault();
    const storageNotes = _.cloneDeep(this.props.storageNotes);
    storageNotes.errors = {};
    this.props.save(storageNotes);
    this.props.history.push(page);
  }


  setSpecies(speciesName, speciesObj) {
    // the species code is generated as id in getItems, that's what we are looking up here
    this.setState({ species: speciesName, speciesCode: speciesObj ? speciesObj.faoCode : speciesName, scientificName: speciesObj?.scientificName }, function () {
      this.onChange(speciesName, 'product');
    });
  }

  onChange(e, name) {
    const { storageNotes, index, scientificName } = this.state;
    const ctch = storageNotes.catches[index];

    const value =  e.target ? e.target.value : null;

    if (name === 'product') {
      ctch[name] = e;
      ctch['scientificName'] = scientificName;
    }else{
      ctch[name] = value;
    }

    this.setState({ storageNotes });
  }

  getErrorsObject(error) {
    let errorObject = {};
    if(!_.isEmpty(error)) {
      Object.keys(error).forEach(key => {
        return errorObject[key] = this.getExtractedError(error[key]);
      });
    }
    return errorObject;
  }

  getExtractedError(error) {
    const {t} = this.props;
    if (!_.isEmpty(error)) {
      const splitDynamicErrors = error.split('-');
      if (splitDynamicErrors.length > 0) {
        return t(splitDynamicErrors[0], {CONSTANT_VALUE: splitDynamicErrors[1]});
      }
    }
    return t(error);
  }

  render = () => {
    const { history, route, match, t } = this.props;
    const { storageNotes, errors, errorsUrl, index, selectedDateOfUnloading } = this.state;
    let { previousUri, progressUri, path } = route;
    const { journey, saveAsDraftUri, title } = route;

    path = path.replace(':productIndex', index);
    previousUri = previousUri.replace(':documentNumber', match.params.documentNumber);

    if (!storageNotes || !storageNotes.addAnotherProduct) return null;
    const ctch = storageNotes.catches[index];
    if (!ctch) return null;
    const {
      product = '',
      commodityCode = '',
      certificateNumber = '',
      productWeight = '',
      weightOnCC = '',
      placeOfUnloading = '',
      transportUnloadedFrom = ''
    } = ctch;

    const errs = (errorsUrl === history.location.pathname && errors) || {}; // only display errors for this URL
    const saveAsDraftFormActionUrl = `/orchestration/api/v1/storageNotes/saveAndValidate?saveAsDraftUrl=${saveAsDraftUri}&c=/create-storage-document/add-product-to-this-consignment/${index}&n=${'/create-storage-document/you-have-added-a-product'}`;

    return (
      <Main>
        <PageTitle title={`${!_.isEmpty(errors) ? 'Error: ' : ''}${t('sdAddProductToThisConsignmentPageTitle')} - ${t(title)}`} />
        <ErrorIsland
          errors={toGovukErrors(this.getErrorsObject(errs))}
          onHandleErrorClick={onHandleErrorClick}
        />
        <GridRow>
          <GridCol>
            <BackLink href={`/orchestration/api/v1/processingStatement/back?n=${previousUri}`}
              onClick={(e) => this.goBack(e, previousUri)}
            >{t('commonBackLinkBackButtonLabel')}</BackLink>
            <Header>{ t('sdAddProductToThisConsignmentPageTitle') }</Header>
          </GridCol>
        </GridRow>
        <Form
          action="/orchestration/api/v1/storageNotes/saveAndValidate"
          currentUrl={`/create-storage-document/${match.params.documentNumber}/add-product-to-this-consignment/${index}`}
          nextUrl={`/create-storage-document/${match.params.documentNumber}/you-have-added-a-product`}
          onSubmit={e => e.preventDefault()}
        >
          <GridRow>
            <GridCol>
              <div className={'input-species-ps properties-group autocomplete-sd ' + ((errors[`catches-${index}-product`]) ? 'error' : '')}>
                <SpeciesAutocomplete
                  hintText={t('sdAddProductToThisConsignmentAutoCompleteHint')}
                  defaultSelectMessage=''
                  defaultValue={product}
                  label={t('sdAddProductToThisConsignmentAutoCompleteLabel')}
                  id={`catches-${index}-product`}
                  name={`catches.${index}.product`}
                  errorName="autoFillError"
                  onChange={this.setSpecies}
                  error={{ errors: [{ targetName: `catches.${index}.product`, text: t(errors[`catches-${index}-product`]) }], autoFillError: t(errors[`catches-${index}-product`]) }}
                  index={index}
                  unauthorised={this.props.storageNotes.unauthorised}
                  htmlFor={`catches.${index}.product`}
                />
                <input type="hidden" name="productCode" value={this.state.species || ''} />
              </div>
              <br />
              <SpeciesDetails exemptFrom={t('sdAddProductToConsignmenStorageDocument')} />
              <br/>
            </GridCol>
          </GridRow>

          <GridRow>
            <GridCol columnTwoThirds>
              <InputField
                style={{ marginTop: '-1em' }}
                id={`catches-${index}-commodityCode`}
                htmlFor={`catches.${index}.commodityCode`}
                meta={{ error: t(errors[`catches-${index}-commodityCode`]), touched: true }}
                input={{ autoComplete: 'off', className: 'formControl mediumInput', value: commodityCode, id: `catches.${index}.commodityCode`, name: `catches.${index}.commodityCode`, onChange: (e) => this.onChange(e, 'commodityCode') }}>{t('commonCommodityCodeLabel')}</InputField>
            </GridCol>
          </GridRow>

          <GridRow>
            <GridCol columnTwoThirds>
              <InputField
                id={`catches-${index}-productWeight`}
                htmlFor={`catches.${index}.productWeight`}
                meta={{ error: t(errors[`catches-${index}-productWeight`]), touched: true }}
                input={{ autoComplete: 'off', className: 'formControl smallInput', value: productWeight, id: `catches.${index}.productWeight`, name: `catches.${index}.productWeight`, onChange: (e) => this.onChange(e, 'productWeight') }}>{t('sdAddProductToThisConsignmentExportWeightLabel')}</InputField>
            </GridCol>
          </GridRow>
          <GridRow>
            <GridCol>
              <Header>{t('sdAddProductToThisConsignmentUKEntryDocumentTitle')}</Header>
              <p style={{ color: '#6f777b' }}>{t('sdAddProductToThisConsignmentUKEntryDocumentParagraph')}</p>
            </GridCol>
          </GridRow>

          <GridRow>
            <GridCol columnTwoThirds>
              <DateFieldWithPicker
                id={`catches-${index}-dateOfUnloading`}
                errors={t(errors[`catches-${index}-dateOfUnloading`])}
                onDateChange={(e)=>this.onChange(e, 'dateOfUnloading')}
                name="dateOfUnloading"
                dateFormat='DD/MM/YYYY'
                date={selectedDateOfUnloading}
                labelText={t('sdAddProductToThisConsignmentDatePickerLabel')}
                labelTextClass='label-product-details-form'
              />
            </GridCol>
          </GridRow>

          <GridRow>
            <GridCol columnTwoThirds>
              <InputField
                id={`catches-${index}-placeOfUnloading`}
                htmlFor={`catches.${index}.placeOfUnloading`}
                style={{ marginTop: '-2em' }}
                meta={{ error: t(errors[`catches-${index}-placeOfUnloading`]), touched: true }}
                hint={t('sdAddProductToThisConsignmentPortHint')}
                input={{ autoComplete: 'off', className: 'formControl mediumInput', value: placeOfUnloading, id: `catches.${index}.placeOfUnloading`, name: `catches.${index}.placeOfUnloading`, onChange: (e) => this.onChange(e, 'placeOfUnloading') }}>{t('sdAddProductToThisConsignmentPortLabel')}</InputField>
            </GridCol>
          </GridRow>

          <GridRow>
            <GridCol columnTwoThirds>
              <InputField
                id={`catches-${index}-transportUnloadedFrom`}
                htmlFor={`catches.${index}.transportUnloadedFrom`}
                meta={{ error: this.getExtractedError(errors[`catches-${index}-transportUnloadedFrom`]), touched: true }}
                hint={t('sdAddProductToThisConsignmentTransportHint')}
                input={{ autoComplete: 'off', className: 'formControl', value: transportUnloadedFrom, id: `catches.${index}.transportUnloadedFrom`, name: `catches.${index}.transportUnloadedFrom`, onChange: (e) => this.onChange(e, 'transportUnloadedFrom') }}>{t('sdAddProductToThisConsignmentTransportLabel')}</InputField>
            </GridCol>
          </GridRow>

          <GridRow>
            <GridCol columnTwoThirds>
              <InputField
                id={`catches-${index}-certificateNumber`}
                htmlFor={`catches.${index}.certificateNumber`}
                meta={{ error: this.getExtractedError(errors[`catches-${index}-certificateNumber`]), touched: true }}
                hint={t('sdAddProductToThisConsignmentDocumentNumberHint')}
                input={{ autoComplete: 'off', maxLength: '54', className: 'formControl largeInput', value: certificateNumber, id: `catches.${index}.certificateNumber`, name: `catches.${index}.certificateNumber`, onChange: (e) => this.onChange(e, 'certificateNumber') }}>{t('commonDocumentNumber')}</InputField>
            </GridCol>
          </GridRow>

          <GridRow>
            <GridCol columnTwoThirds>
              <InputField
                id={`catches-${index}-weightOnCC`}
                htmlFor={`catches.${index}.weightOnCC`}
                meta={{ error: t(errors[`catches-${index}-weightOnCC`]), touched: true }}
                input={{ autoComplete: 'off', className: 'formControl smallInput', value: weightOnCC, id: `catches.${index}.weightOnCC`, name: `catches.${index}.weightOnCC`, onChange: (e) => this.onChange(e, 'weightOnCC') }}>{t('sdAddProductToThisConsignmentWeightOnDocumentLabel')}</InputField>
            </GridCol>
          </GridRow>

          <GridRow>
            <SaveAsDraftButton formactionUrl={saveAsDraftFormActionUrl} onClick={this.onSaveAsDraft} />
            <input type="hidden" name="journey" value={journey} />
            <input type="hidden" name="currentUri" value={path} />
            <ContinueButton id="continue" onClick={() => this.save(`/create-storage-document/${match.params.documentNumber}/you-have-added-a-product`)}>
              {t('commonContinueButtonSaveAndContinueButton')}
            </ContinueButton>
          </GridRow>
        </Form>
        <BackToProgressLink
          progressUri={progressUri}
          documentNumber={match.params.documentNumber}
        />
        <HelpLink journey={journey} />
      </Main>
    );
  };
}

function mapStateToProps(state) {
  return {
    storageNotes: state.storageNotes
  };
}

function loadData(store) {
  return store.dispatch(getAllFish())
    .then(store.dispatch(getStorageNotesFromRedis(this.documentNumber)));
}

export const component = withRouter(
  connect(
    mapStateToProps, {
    save: saveStorageNotes,
    saveToRedis: saveStorageNotesToRedis,
    getFromRedis: getStorageNotesFromRedis,
    clear: clearStorageNotes,
    getAllFish
  }
  )(withTranslation() (CatchDetailsPage))
);

export default {
  loadData,
  component: PageTemplateWrapper(component)
};
