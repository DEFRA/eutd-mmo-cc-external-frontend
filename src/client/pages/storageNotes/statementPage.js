import React, { Fragment } from 'react';
import { withRouter } from 'react-router';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';

import {
  Main,
  Header,
  Panel,
  GridRow,
  GridCol
} from 'govuk-react';

import {
  clearStorageNotes, clearTransportDetails,
  saveStorageNotes, saveStorageNotesToRedis
} from '../../actions';
import {
  getCompletedDocument
} from '../../actions/document.actions';
import PageTitle from '../../components/PageTitle';
import PageTemplateWrapper from '../../components/PageTemplateWrapper';
import HelpLink from '../../components/HelpLink';
import { monitorEvent } from '../../actions/monitor.actions';
import {withTranslation} from 'react-i18next';

class StatementPage extends React.Component {

  hasRequiredData() {
    const { completedDocument } = this.props;

    return completedDocument && completedDocument.documentNumber && completedDocument.documentUri && completedDocument.documentNumber.includes('-SD-');
  }

  async componentDidMount() {
    const documentNumber = this.props.match.params['documentNumber'];

    this.props.clear({ catches: [{}], storageFacilities: [{}], pdf: this.props.storageNotes.pdf });
    this.props.clearTransportDetails();

    await this.props.getCompletedDocument(documentNumber);

    if (!this.hasRequiredData()) {
      this.props.history.push('/create-storage-document/storage-documents');
    }
  }

  render() {
    const {route, completedDocument, t} = this.props;
    const documentNumber = (completedDocument && completedDocument.documentNumber) ? completedDocument.documentNumber : 'not found';
    const uri = (completedDocument && completedDocument.documentUri) ? completedDocument.documentUri :  'not found';

    const buildUrl = (docUrl) => {
      if(docUrl.indexOf('/') !== -1) {
        const splitUrlLength = docUrl.split('/').length;
        const splitUrl = docUrl.split('/')[splitUrlLength - 1];
        const docId = splitUrl.split('?')[0];
        return `/pdf/export-certificates/${docId}`;
      }
      return `/pdf/export-certificates/${docUrl}`;
    };

    const monitorViewEvent = doc => {
      const journey = this.props.route.journey;
      this.props.monitorEvent(doc, journey);
    };

    return (
      <Main>
        <PageTitle title={`${t('sdStatementPageTitle')} - ${t('sdCommonTitle')}` }/>
        <GridRow>
          <GridCol>
            <Panel
              className="submitted-certificate-bg"
              panelTitle={t('sdCreatedPanelTitle')}
              panelBody={
                <Fragment>
                  {t('sdCreatedPanelBody')}
                  <br />
                  <strong id="documentNumber">{documentNumber}</strong>
                </Fragment>
              }
            />
          </GridCol>
        </GridRow>
        <GridRow>
          <GridCol>
            <Header level={2} size="MEDIUM">{t('sdCreatedNextStepsHeader')}</Header>
            <ol className="list list-number">
              <li>
                <strong><a rel="noopener noreferrer" onClick={monitorViewEvent.bind(this, documentNumber)} href={buildUrl(uri)}>{t('sdCreatedDownloadLink')}</a>. </strong>
                {t('sdCreatedDownloadDocumentNotesSubHeading')}
                <ul className="list list-bullet">
                  <li>{t('sdCreatedDownloadDocumentNotesFirefox')}</li>
                  <li>{t('sdCreatedDownloadDocumentNotesMobile')}</li>
                </ul>
              </li>
              <li>
                <strong>{t('sdCreatedEmailToImporter')}. </strong>
                <span>{t('sdCreatedEmailToImporterText')}</span>
              </li>
            </ol>
            <p>
              <Link to={'/create-storage-document/storage-documents'}>{t('sdCreatedViewOrCreateNewLink')}</Link>
            </p>
            <HelpLink journey={route.journey}/>
          </GridCol>
        </GridRow>
      </Main>
    );
  }
}

function mapStateToProps(state) {
  return {
    storageNotes: state.storageNotes,
    documents: state.documents,
    completedDocument: state.completedDocument,
    monitorEvent: monitorEvent
  };
}

function loadData(store) {
  if (Object.keys(this.queryParams).length !== 0) {
    store.dispatch(saveStorageNotes({ pdf: { uri: this.queryParams.uri, documentNumber: this.queryParams.documentNumber } }));
  }
}

export const component = withRouter(connect(mapStateToProps,
  {
    save: saveStorageNotes,
    saveToRedis: saveStorageNotesToRedis,
    clear: clearStorageNotes,
    monitorEvent,
    clearTransportDetails,
    getCompletedDocument
    }
  )(withTranslation()(StatementPage))
);

export default {
  loadData,
  component: PageTemplateWrapper(component)
};
