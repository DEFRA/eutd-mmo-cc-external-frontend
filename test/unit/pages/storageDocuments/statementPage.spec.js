import { mount } from 'enzyme';
import * as React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { render } from '@testing-library/react';
import * as StatementPage from '../../../../src/client/pages/storageNotes/statementPage';

describe('Statement Page', () => {
  const props = {

    route: {
      title: 'Create a UK storage document for exports',
      previousUri: '/export-certificates/add-exporter-details',
      nextUri: '/add-landings',
      path: ':documentNumber/what-are-you-exporting',
      saveAsDraftUri: '/create-storage-document/storage-document',
      journey: 'processingStatement'
    },
    completedDocument: {
      documentNumber: 'GBR-2021-SD-7FF42686F',
      documentStatus: 'COMPLETE',
      documentUri: '_75d4eda9-2f5a-46d4-9324-9fe28915575d.pdf'
    }
  };
  const mockStore = configureStore([thunk]);


  const store = mockStore({
    completedDocument: {
      documentNumber: 'GBR-2021-SD-7FF42686F',
      documentStatus: 'COMPLETE',
      documentUri: '_75d4eda9-2f5a-46d4-9324-9fe28915575d.pdf'
    },
    errors: {}
  });

  const wrapper = mount(
    <Provider store={store}>
      <MemoryRouter>
        <StatementPage.default.component  {...props} />
      </MemoryRouter>
    </Provider>);

  it('should render complete page', () => {
    expect(wrapper).toBeDefined();
  });

  it('should render complete title with document number', () => {
    expect(wrapper.find('StyledTitle h2').text()).toBe('The storage document has been created');
    expect(wrapper.find('StyledBody #documentNumber').text()).toEqual('GBR-2021-SD-7FF42686F');
  });

  it('should should create snapshot of the UI', () => {
    const { container } = render(wrapper);
    expect(container).toMatchSnapshot();
  });
});