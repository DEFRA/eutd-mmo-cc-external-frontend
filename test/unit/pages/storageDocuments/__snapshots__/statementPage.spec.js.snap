// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Statement Page should should create snapshot of the UI 1`] = `
<div>
  <div
    class="css-1l99ixu-OuterContainer e4u1u2g0"
  >
    <div
      class="css-n57fe4-InnerContainer e4u1u2g1"
    >
      <div
        class="css-1lzr9ic-StyledRow e4u1u2g0"
      >
        <div
          class="css-1w80l5h-StyledColumn e4u1u2g0"
        >
          <div
            class="submitted-certificate-bg e1vcssft0 css-1y5yjt5-StyledPanel-StyledHoc e4u1u2g0"
          >
            <h2
              class="css-em1gqr-StyledTitle e4u1u2g1"
            >
              The storage document has been created
            </h2>
            <div
              class="css-1l9w67v-StyledBody e4u1u2g2"
            >
              Storage statement number
              <br />
              <strong
                id="documentNumber"
              >
                GBR-2021-SD-7FF42686F
              </strong>
            </div>
          </div>
        </div>
      </div>
      <div
        class="css-1lzr9ic-StyledRow e4u1u2g0"
      >
        <div
          class="css-1w80l5h-StyledColumn e4u1u2g0"
        >
          <h2
            class="e1vcssft0 css-1wlsbnr-StyledHeader-StyledHoc e4u1u2g0"
          >
            What you need to do next
          </h2>
          <ol
            class="list list-number"
          >
            <li>
              <strong>
                <a
                  href="/pdf/export-certificates/_75d4eda9-2f5a-46d4-9324-9fe28915575d.pdf"
                  rel="noopener noreferrer"
                >
                  Download the storage document
                </a>
                . 
              </strong>
              Do not amend the document. Please note:
              <ul
                class="list list-bullet"
              >
                <li>
                  if you are using Firefox as an internet browser, please ensure JavaScript is enabled in order to view and download the certificate
                </li>
                <li>
                  if you are using a mobile device, please ensure you have installed a PDF viewer
                </li>
              </ul>
            </li>
            <li>
              <strong>
                Email the storage document to the importer
                . 
              </strong>
              <span>
                It is the importer's responsibility to submit it to the import control authority where your export will enter the EU. The importing authority will complete and sign their section of the document at the Border Inspection Post (BIP).
              </span>
            </li>
          </ol>
          <p>
            <a
              href="/create-storage-document/storage-documents"
            >
              View completed storage documents or create a new export document
            </a>
          </p>
          <div
            class="css-1lzr9ic-StyledRow e4u1u2g0"
          >
            <div
              class="css-1w80l5h-StyledColumn e4u1u2g0"
            >
              <hr
                class="help-link"
              />
              <h2
                class="e1vcssft0 css-162lx7h-StyledHeader-StyledHoc e4u1u2g0"
              >
                Need help?
              </h2>
              <a
                href="undefined"
                rel="noopener noreferrer"
                target="_blank"
              >
                Get help exporting fish from the UK (gov.uk)
                 
                <span
                  class="govuk-visually-hidden"
                >
                  (opens in new tab)
                </span>
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
